src/conn/conn_ckpt.c

```diff
@@ -78,14 +94,20 @@ __ckpt_server(void *arg)
 	conn = S2C(session);
 	wt_session = (WT_SESSION *)session;
 
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(conn, WT_CONN_SERVER_CHECKPOINT)) {
+	for (;;) {
+		if (!__ckpt_server_run_chk(session))
+			break;
```

Comment:
> Why do we need to check before calling __wt_cond_wait? It's going to check before going to sleep...

I removed the call of the `XXX_run_chk()` functions before the call to `__wt_cond_wait()`.